sdk/js/index.ts

```diff
@@ -312,17 +312,19 @@ export default class StreamManager {
     });
   }
 
-  async fetchAllLogs(callback: (logs: Log[]) => void, fromBlock?: bigint) {
-    try {
-      const logs = await this.publicClient.getContractEvents({
+  fetchAllLogs(callback: (logs: Log[]) => void, fromBlock?: bigint): void {
```

Comment:
was hoping to refactor like this
```suggestion
  fetchAllLogs(handleStream: (log: Log) => void, fromBlock?: bigint): void {
```